File: 05-functions/my-code/202212/lab05.py
from rpg import *


def main() -> None:

	health, attack, defense = map(int, input().split())
	player = RPGChar("Sarah", health, attack, defense, "a")

	health, attack, defense = map(int, input().split())
	enemy = RPGChar("O clone", health, attack, defense, "o")

	seed = int(input())

	rpg = RPG(player, enemy, seed)

	while not rpg.battle_ended:
		rpg.set_user_target()

		rpg.user.last_move = input()
		rpg.user.execute_move(rpg)

		rpg.check_battle_end()
		rpg.player_turn = not rpg.player_turn


if __name__ == "__main__":
	main()

File: 05-functions/my-code/202212/lab05.c
#include <stdio.h>
#include <stdlib.h>
#include "rpg.h"

int main(void)
{
	struct rpg_char *player, *enemy;
	struct rpg *g;
	int seed;

	player = create_character("Sarah", 'a');
	enemy = create_character("O clone", 'o');

	scanf("%d", &seed);

	g = create_rpg(player, enemy, seed);

	while (!g->battle_ended) {
		set_user_target(g);

		scanf("%s", g->user->last_move);
		execute_move(g);

		check_battle_end(g);
		g->player_turn = !g->player_turn;
	}

	free(player);
	free(enemy);
	free(g);

	return 0;
}
